selfservice/hook/web_hook.go

```diff
@@ -209,7 +210,15 @@ func (e *WebHook) ExecutePostRegistrationPostPersistHook(_ http.ResponseWriter,
 	if gjson.GetBytes(e.conf, "can_interrupt").Bool() || gjson.GetBytes(e.conf, "response.parse").Bool() {
 		return nil
 	}
-	return otelx.WithSpan(req.Context(), "selfservice.hook.WebHook.ExecutePostRegistrationPostPersistHook", func(ctx context.Context) error {
+
+	// We want to decouple the request from the hook execution, so that the hooks still execute even
+	// if the request is canceled.
+	var cancel context.CancelFunc
+	ctx := trace.ContextWithSpan(context.Background(), trace.SpanFromContext(req.Context()))
+	ctx, cancel = context.WithTimeout(ctx, time.Second*30)
```

Comment:
This changes the retry behavior. Before, we had three retries of 30 sec each, now we have a total of 30sec including all retries.

Reply:
Ah good catch! The reason I put some timeout here is because the Jsonnet VM will try to spin up a process until the parent process is cancelled, and we prob want some general timeout here. 5 minutes? WDYT?

```suggestion
	ctx, cancel = context.WithTimeout(ctx, 5*time.Minute)
```

Reply:
Could we use the same context/cancel logic for all web hook types?

Reply:
Most of the web hooks need to be executed in the request context and we rely on their answer for further processing. For post-persist hooks and async hooks, we don't rely on the answer, so we can execute them in another context.

Reply:
Yeah, interrupting post-persist hooks don't actually make a ton of semantic sense. We should probably rework the config to deprecate them.